### PR TITLE
add API to get SSRC and audio levels

### PR DESCRIPTION
**Pull request by fluffy**
Fixes https://github.com/w3c/webrtc-pc/issues/4 and https://github.com/w3c/webrtc-pc/issues/6 and https://github.com/w3c/webrtc-pc/issues/276

### CODE REVIEW

**Comment by fluffy** [comment]
Need to indicate which bugs this fixes in the PR

File: webrtc.html [hunk 1]
```diff
@@ -2948,8 +2948,57 @@
...
             In the case where no a=mid attribute is present in the remote
             description, a random value will be generated.
-          </dd>
-        </dl>
+	  </dd>
+
+	  <dt> sequence&lt;RTCRtpContributingSource&gt;
+	    getContributingSources ()  </dt>
+	  <dd> This returns a RTCRtpContributingSource for each unique
+	  CSRC or SSRC received by this RTCRtpReceiver in the last 10
```
Comment (ekr) [inline webrtc.html]: s/This r/R/

File: webrtc.html [hunk 2]
```diff
@@ -2948,8 +2948,57 @@
...
+	  seconds.  </dd>
+	  
+	</dl>
+
+	<p> The RTCRtpContributingSource objects contian information
+	provided about a given contribution source included time most
```
Comment (ekr) [inline webrtc.html]: s/contian/contain/ ->

File: webrtc.html [hunk 3]
```diff
@@ -2948,8 +2948,57 @@
...
+	   <dt> readonly attribute unsigned long source</dt>
+	   <dd> The CSRC or SSRC value of the contributing source. </dd>
+
+	   <dt> readonly attribute byte? audioLevel</dt>
+	   <dd>
+	     The audio level contained in the last RTP packet received
+	   from the this source. If
+	   the source was a CSRC, this will be the value defined as
+	   the level in
+	   [[!RFC6464]] while if the source was set from a CSRC, this will
```
Comment (ekr) [inline webrtc.html]: What's the difference between being from a CSRC and being set from a CSRC.

File: webrtc.html [hunk 4]
```diff
@@ -2948,8 +2948,57 @@
...
+	   be the value as defined in level in [[!RFC6465]]. Those
+	   specification defile this level as a value from 0 to -127
+	   representing the audio level in decibels relative to the
```
Comment (ekr) [inline webrtc.html]: s/defile/define/

Comment (ekr) [inline webrtc.html]: s/value/integral value/

File: webrtc.html [hunk 5]
```diff
@@ -2948,8 +2948,51 @@
...
+
+	  <dt> sequence&lt;RTCRtpContributingSource&gt;
+	    getContributingSources ()  </dt>
+	  <dd> Returns a RTCRtpContributingSource for each unique
+	  CSRC or SSRC received by this RTCRtpReceiver in the last 10
+	  seconds.  </dd>
+	  
+	</dl>
+
+	<p> The RTCRtpContributingSource objects contians information
```
Comment (jan-ivar) [inline webrtc.html]: missed one.

Comment (fluffy) [inline webrtc.html]: thanks - fixed

**Comment by fluffy** [comment]
Folks, I think this represents all the comments from interim meeting and is ready to merge. (Perhaps there are some more typos but it is close enough to merge)

File: webrtc.html [hunk 6]
```diff
@@ -2948,8 +2948,51 @@
...
+	received, the entries are updated for the SSRC of the RTP
+	packet and for any CSRC contained in it. </p>
+
+	 <dl class="idl" title="interface RTCRtpContributingSource">
+
+	   <dt> readonly attribute DOMHiResTimeStamp timestamp</dt>
+
+	   <dd> Time of reception of the most recent RTP packet
+	   containing the contributing source. </dd>
+
```
Comment (dontcallmedom) [inline webrtc.html]: the actual definition of what the timestamp refers to was discussed [on the list](https://lists.w3.org/Archives/Public/public-webrtc/2015Sep/0058.html); shouldn't this be reflected here? (either as "to be determined" or with a specific proposal for review)

Comment (fluffy) [inline webrtc.html]: I agree - I will update the PR to reflect this.

**Comment by alvestrand** [comment]
Met with approval in Seattle. Should be merged.

**Comment by alvestrand** [comment]
Some updates on the definition of time are needed.

**Comment by fluffy** [comment]
This is ready to merge

**Comment by alvestrand** [comment]
@fluffy can you go on the mailing list and reply to the "Potential problem with DOMHghResTimeStamp" thread and say that you think it is unproblematic to use this time differently from everything else in the web platform that uses it?
(in all other usages, DOMHighResTimeStamp are relative to document load time. Bernard's suggested that in this case, it should be relative to the Unix epoch. It's an offset, not an absolute time. Yes, this is a problem with stats too.)

**Comment by fluffy** [comment]
I think we should use existing high res time exactly as defined - that is to say it would not be relative to Unix epoch. I think that is what we agreed to at meeting so if someone wants to make it some other sort of time, I'd rather merge this inn then look at PR that change that

**Comment by martinthomson** [comment]
I agree with @fluffy.  Epoch can be document load time.

**Comment by aboba** [comment]
Found some typos/inconsistencies, and created an updated PR (318):
https://github.com/w3c/webrtc-pc/pull/318

Take a look and see if this gets us closer.

**Comment by alvestrand** [comment]
@aboba do you think you can update this PR instead? Having more PRs for the same thing confuses the trail.
@fluffy @martinthomson can you state your opinion on the list? I'm always hesitant to declare consensus when one opinion is stated on the list and another one is stated in a Github comment thread.

**Comment by fluffy** [comment]
I merged @aboba PR as it looked good

**Comment by aboba** [comment]
Thanks, Cullen.

One question.

For the case where the RTP packet contained CSRCs and a mixer-client
extension, but no client-mixer extension, is the intent for the browser to
include an entry for the SSRC and audio level as though the client-mixer
extension was present?  Wasn't sure if this was only relevant for the P2P
case (where there would be no CSRCs, just SSRC and the client-mixer
extension).

**Comment by aboba** [comment]
Noticed a typo: 

+    information about a given contributing source, including the time
+    the most recent time a packet was received from the source. The

[BA] s/including the time/including/

**Comment by stefhak** [comment]
I think we should merge this PR now, and have it part of a new Ed's draft for wider review.

**Pull request merged**